frontend/src/components/nearAuth/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { WalletConnection } from "near-api-js"
+import { navigate } from "gatsby"
+import React, { createContext, useContext, useState, useEffect, ReactNode } from "react"
+import { getNearWallet, signIn, signOut } from "../../near-wallet"
+
+type IWallet = {
+    authenticated: boolean
+    login: () => Promise<void>
+    logout: () => void
+}
+
+const NEARContext = createContext<IWallet | undefined>(undefined)
+
+export function useWallet() {
+    return useContext(NEARContext)
+}
+
+export function WalletProvider(props: { children: ReactNode }) {
+    const [wallet, setWallet] = useState<WalletConnection | undefined>(undefined)
+
+    const [authenticated, setAuthenticated] = useState<boolean>(false)
+
+    async function login() {
+        await signIn(wallet!)
+    }
+
+    function logout() {
+        signOut(wallet!)
+        setAuthenticated(wallet!.isSignedIn())
+    }
+
+    useEffect(function () {
+        getNearWallet().then(w => {
```

Comment:
would be a tiny bit cleaner if we could prioritize `async...await` over `then()` as the former is becoming the norm (if not already is) :)

Reply:
I tried. `useEffect` does not like async function. Technically it will work, but Typescript is not happy with me.